src/main/java/teammates/ui/controller/FeedbackSubmissionEditPageData.java

```diff
@@ -180,7 +185,57 @@ public void setRegisterMessage(String registerMessage) {
 
         return result;
     }
-    
+
+    public void filterSessionQuestionBundle() {
+
+        Set<String> notDisplayedInstructorEmails = getNotDisplayedInstrutorEmails();
+
+        // Remove instructors who are not displayed to students
+        for (FeedbackQuestionAttributes question : bundle.questionResponseBundle.keySet()) {
+            
+            if (question.recipientType != FeedbackParticipantType.INSTRUCTORS) {
+                continue;
+            }
+            
+            Map<String, String> recipients = bundle.recipientList.get(question.getId());
+            List<FeedbackResponseAttributes> responses = bundle.questionResponseBundle.get(question);
+
+            for (String instrEmail : notDisplayedInstructorEmails) {
+                
+                if (recipients.containsKey(instrEmail)) {
+                    recipients.remove(instrEmail);
+                }
+
+                // Remove the responses if they have been stored already
+                Iterator<FeedbackResponseAttributes> iterResponse = responses.iterator();
+
+                while (iterResponse.hasNext()) {
+                    
+                    FeedbackResponseAttributes response = iterResponse.next();
+
+                    if (response.recipientEmail.equals(instrEmail)) {
+                        iterResponse.remove();
+                    }
+                }
+            }
+        }
+    }
+
+    private Set<String> getNotDisplayedInstrutorEmails() {
+
+        Logic logic = new Logic();
```

Comment:
@unyoungwax, if there's any processing, you might need to move it into FSLogic itself